packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -231,6 +231,117 @@ func GetBlockProofStatus(
 	}, nil
 }
 
+// BatchGetBlockProofStatus checks whether the batch of L2 block still need new proofs or new contests.
+// Here are the possible status:
+// 1. No proof on chain at all.
+// 2. A valid proof has been submitted.
+// 3. An invalid proof has been submitted, and there is no valid contest.
+// 4. An invalid proof has been submitted, and there is a valid contest.
+func BatchGetBlockProofStatus(
+	ctx context.Context,
+	cli *Client,
+	ids []*big.Int,
+	proverAddress common.Address,
+	proverSetAddress common.Address,
+) ([]*BlockProofStatus, error) {
+	ctxWithTimeout, cancel := CtxWithTimeoutOrDefault(ctx, defaultTimeout)
+	defer cancel()
+
+	var (
+		parentHashes = make([][32]byte, len(ids))
+		parents      = make([]*types.Header, len(ids))
+		blockIDs     = make([]uint64, len(ids))
+		result       = make([]*BlockProofStatus, len(ids))
+	)
+	// Get the local L2 parent header.
+	for i, id := range ids {
+		parent, err := cli.L2.HeaderByNumber(ctxWithTimeout, new(big.Int).Sub(id, common.Big1))
+		if err != nil {
+			return nil, err
+		}
+		parentHashes[i] = parent.ParentHash
+		parents[i] = parent
+		blockIDs[i] = id.Uint64()
+	}
+
+	// Get the transition state from TaikoL1 contract.
+	transitions, err := cli.TaikoL1.GetTransitions(
+		&bind.CallOpts{Context: ctxWithTimeout},
+		blockIDs,
+		parentHashes,
+	)
+	if err != nil {
+		return nil, err
+	}
+	for i, transition := range transitions {
```

Comment:
Do we need check `len(transitions) == len(ids)`?

Reply:
No need, have test it. Returned empty transition when can not find actual transition by id.